lib/gscraper/account/users.ex

```diff
@@ -37,4 +78,30 @@ defmodule Gscraper.Account.Users do
   def change_user(%User{} = user, attrs \\ %{}) do
     User.changeset(user, attrs)
   end
+
+  @doc """
+  Authenticate a user with provided credentials.
+
+  ## Examples
+
+      iex> authenticate_user(valid_username, valid_password)
+      {:ok, %User{}}
+
+      iex> authenticate_user(valid_username, invalid_password)
+      {:error, :invalid_credentials}
+
+  """
+  def authenticate_user(username, plain_password) do
+    case find_by_username(username) do
+      nil ->
+        {:error, :invalid_credentials}
```

Comment:
Maybe we could recheck if we should apply [no_user_verify](https://hexdocs.pm/argon2_elixir/Argon2.html#no_user_verify/1) to prevent timing attacks 💡

Reply:
It's good to know that 🙏 I made the change in d8e7526